server/Arcadia.Assistant.Web/Download/DownloadActor.cs

```diff
@@ -0,0 +1,239 @@
+
+        public DownloadActor(
+            IDownloadApplicationSettings downloadApplicationSettings,
+            IHttpClientFactory httpClientFactory,
+            string basePath)
+        {
+            this.downloadApplicationSettings = downloadApplicationSettings;
+            this.httpClientFactory = httpClientFactory;
+            this.basePath = basePath;
+            Context.System.Scheduler.ScheduleTellRepeatedly(
+                TimeSpan.Zero,
+                TimeSpan.FromMinutes(downloadApplicationSettings.DownloadBuildIntervalMinutes),
+                this.Self,
+                RefreshApplicationBuilds.Instance,
+                this.Self);
+        }
+
+        public IStash Stash { get; set; }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetLatestApplicationBuildPath msg:
+                    this.RespondLatestApplicationBuildPath(msg.ApplicationType);
+                    break;
+
+                case RefreshApplicationBuilds msg:
+                    this.Self.Tell(msg);
+                    this.Become(this.DownloadingApplicationBuilds);
+                    break;
+
+                default:
+                    this.Unhandled(message);
+                    break;
+            }
+        }
+
+        private void DownloadingApplicationBuilds(object message)
+        {
+            switch (message)
+            {
+                case RefreshApplicationBuilds _:
+                    this.DownloadApplicationBuilds()
+                        .PipeTo(
+                            this.Self,
+                            success: () => RefreshApplicationBuildsFinish.Instance,
+                            failure: err => new RefreshApplicationBuildsFinish(err.Message));
+                    break;
+
+                case RefreshApplicationBuildsFinish msg:
+                    if (msg.Message != null)
+                    {
+                        this.logger.Warning(msg.Message);
+                    }
+
+                    this.Become(this.DefaultState());
+
+                    break;
+
+                default:
+                    this.Stash.Stash();
+                    break;
+            }
+        }
+
+        private UntypedReceive DefaultState()
+        {
+            this.Stash.UnstashAll();
+            return this.OnReceive;
+        }
+
+        private void RespondLatestApplicationBuildPath(
+            GetLatestApplicationBuildPath.ApplicationTypeEnum applicationType)
+        {
+            var buildPath = applicationType == GetLatestApplicationBuildPath.ApplicationTypeEnum.Android
+                ? this.androidApplicationBuildPath
+                : this.iosApplicationBuildPath;
+            this.Sender.Tell(new GetLatestApplicationBuildPath.Response(buildPath));
+        }
+
+        private async Task DownloadApplicationBuilds()
+        {
+            var androidDownloadBuildTask = this.DownloadBuild(
```

Comment:
I don't like that you download it each time even if nothing changed. Probably save latest build id in a field? If you do that you'd probably better have separate 'download' actor that just downloads thing and drops it.